File: haskell/Fizz1.hs
import Control.Monad

main = fizzBuzz 100

fizzBuzz :: Integer -> IO()
fizzBuzz num = do
  forM_ [1..num] $ \i -> do
    if (i `mod` 3 == 0)
    then
      if (i `mod` 5 == 0)
      then
        print "FizzBuzz"
      else
        print "Fizz"
    else
      if (i `mod` 5 == 0)
      then
        print "Buzz"
      else
        print $ show i

File: ocaml/Fizz1.ml
for i = 1 to 100 do
	if i mod 15 = 0 then print_string "FizzBuzz\n"
	else if i mod 3 = 0 then print_string "Fizz\n"
	else if i mod 5 = 0 then print_string "Buzz\n"
	else(
		print_int i; print_newline ()
	)
done